src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,95 @@
-import { useState } from 'react';
+import { FormEvent, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+interface NewMovieProps {
+  onAdd: (arg0: Movie) => void,
+}
+
+export const NewMovie = ({ onAdd }: NewMovieProps) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imageUrl, setImageUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const canSubmit = title && imageUrl && imdbUrl && imdbId;
+
+  const movie: Movie = {
+    title,
+    description,
+    imgUrl: imageUrl,
+    imdbUrl,
+    imdbId,
+  };
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>, newMovie: Movie) => {
+    event.preventDefault();
+    onAdd(newMovie);
+    setCount(count + 1);
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={(event) => {
+        handleSubmit(event, movie);
+      }}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={(input) => {
+          setTitle(input);
+        }}
         required
       />
 
       <TextField
         name="description"
         label="Description"
-        value=""
+        value={description}
+        onChange={(input) => {
+          setDescription(input);
```

Comment:
Try this again, maybe this time it will work 🙇 
```suggestion
        onChange={setDescription};
```